### DISCUSSION TITLE
Batch runner doesn't set initial condition / lost initial values

### DISCUSSION THREAD

**Discussion by karenlaw** [mesa/mesa#1629] [Q&A]
I have a simple simulation that runs correctly by itself or with visualization, but when I try to run it with batch runner, the initial condition is lost sometime and the run failed. Any help is welcome. I am very lost. This is run in __init__ in model

        ...
        for p in self.input_p:
            p_pos = (self.i, p['pos'][1] + self.offset)
            pa = P(p['name'], 
                                  p_pos, 
                                  p['s']['name'],
                                  self)
            if not self.grid.is_cell_empty(p_pos):
                for agent in self.grid.iter_cell_list_contents(p_pos):
                    self.grid.remove_agent(agent)
            self.grid.place_agent(pa, p_pos)
            self.pas.append(pa)
            print("pa>",pa)
        
        for p in self.pas:
            print(p)
         ...
   
Here is part of the print out:
...
pa> pa68:(4, 2)
pa> pa69:(4, 1)
pa> pa70:(4, 0)
pa1:None
pa2:None
pa3:None
...

You can see when I create the variable, it has the value for pos, but when I print it out after I created them, the pos value is gone. This only happens when I use batch_run. Anyone has any idea where I should look for answers or what could be going on?

**Comment by tpike3** [comment]
When you use batchrunner, it is using multiple processors, but only one processor will output to the screen. Due these challenges debugging with multiple processors is very difficult. It is best to only use 1 processor for debugging and verifying your code, and then use multiple processors when running to get your results and conduct parameter sweeps.
